util/php-dictionary-gen/php-dictionary-creator.sh

```diff
@@ -0,0 +1,485 @@
+#!/bin/bash
+#
+# This is a utility script to create wordlists for later use by the
+# OWASP ModSecurity Core Rule Set.
+#
+# The scripts extracts function names out of the PHP source code and
+# filters them into different categories.
+#
+
+IFS=$'\n\t'
+
+# --------------------------------------------------
+# Initialization
+# --------------------------------------------------
+
+VERBOSE=0
+ERROR=0
+
+MYDATE=$(date +"%Y-%m-%d")
+MYDATE_SECONDS=$(date +"%s")
+
+AGE_LIMIT=30
+FREQUENCY_LIMIT=90000
+
+RULES="933150 933151 933161"
+RULES_CMDLINE=""
+
+PHP_REPO=""
+PHP_REPO_CMDLINE=""
+PHP_REPO_GITHUB="https://github.com/php/php-src"
+
+SPELL_PATH="../fp-finder/spell.sh"
+SPELL_PATH_CMDLINE=""
+
+
+TMP_PHP_FUNCTIONS_FREQUENCIES=$(mktemp)
+TMP_PHP_FUNCTIONS_FREQUENCIES_ERRORS=$(mktemp)
+PHP_FUNCTIONS_FREQUENCIES_CMDLINE=""
+
+TMP_PHP_FUNCTIONS=$(mktemp)
+TMP_ENGLISH_WORDS=$(mktemp)
+TMP_PHP_FUNCTIONS_FREQUENT=$(mktemp)
+TMP_PHP_FUNCTIONS_RARE=$(mktemp)
+TMPDIR=$(mktemp -d)
+
+trap 'rm -rf $TMP_PHP_FUNCTIONS $TMP_PHP_FUNCTIONS_FREQUENCIES  $TMP_PHP_FUNCTIONS_FREQUENT $TMP_PHP_FUNCTIONS_RARE $TMP_ENGLISH_WORDS $TMPDIR' INT TERM EXIT
```

Comment:
I wouldn't do this. You already have a date check in place. If the script fails at some point (after 2 hours because of a network error, for example) you don't want to rerun the full script but pick up where you left off. You should be able to pick up the work by skipping the entries that are young enough.

You could still add a `--clean` switch if you want.

Reply:
All these files would still be not used in a second run, being random temp files generated.
We can start again from where the script failed (or from the last successful run) by setting an explicit path where to store the frequency list.
For context, I'm running the script with the following line: `./php-dictionary-creator.sh --frequencylist ~/frequencylist.txt`. Doing so, entries are individually stored into `frequencylist.txt` and kept for the next runs.

To automate it I could make `TMP_PHP_FUNCTIONS_FREQUENCIES` a fixed path, but then I would at least keep it in under the `util/` folder, and add it to `.gitignore`. No strong feelings about it, but I think that it makes sense to just be aware of where you store the frequency list file

Reply:
You could simply make `-f` a mandatory option / positional parameter. If there's no file, it will be created, otherwise it will be used like now. I think that would make it more clear how to use the script and will prevent the issue.

Reply:
I did not report back here. Sorry.
By default now `util/php-dictionary-gen/frequencylist.txt` is used if no other frequency list file is specified via `--frequencylist`. Being still meant to be only a local file it has also been added to `.gitignore`